src/test/ui/statics/uninhabited-static.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(never_type)]
+#![deny(uninhabited_static)]
+
+enum Void {}
+extern {
+    static VOID: Void; //~ ERROR static of uninhabited type
+    //~| WARN: previously accepted
+    static NEVER: !; //~ ERROR static of uninhabited type
```

Comment:
does the diagnostic for a regular `static NEVER: ! = panic!();` change with this PR?

Reply:
Yes, looks like it does (see the test change I just pushed)... would you prefer it to only apply to foreign items?

Reply:
No, having tests for it is good, I'm mostly suprised we don't have any other tests for this